app/models/tag.rb

```diff
@@ -165,4 +167,10 @@ def remove_from_managed_filters
   def name_path
     @name_path ||= name.sub(%r{^/[^/]*/}, "")
   end
+
+  def name_uniqueness
```

Comment:
It looks like `validates :name, :unique_within_region => true` would cover this, right?

Reply:
Also, we need a migration to rename any existing duplicates, right?

Reply:
NICE: `:unique_within_region => true`

Reply:
There was already a uniqueness constraint on tags before. At least those created through classifications.
This change moved the uniqueness from classification_tag over to tags (seemed to make more sense)

While I don't think we need a migration, it does sound like the safe thing to do.